### ISSUE TITLE
ENH: explore() extend _MAP_KWARGS to include all folium / leaflet initialisation parameters

### ISSUE THREAD

**Issue by rraymondgh** [geopandas/geopandas#2313]
#### Is your feature request related to a problem?

**explore.py** contains list `_MAP_KWARGS` that is used to enumerate parameters set to `folium.Map()`.  It is incomplete - extend to include all options

See: https://stackoverflow.com/questions/70795378/how-to-disable-zoom-and-pan-in-a-geopandas-map-created-with-explore/70799430#70799430

#### Describe the solution you'd like

extend or auto-generate all valid **kwargs** to `folium.map()`

#### API breaking implications

None

#### Describe alternatives you've considered

Have directly patched `geopandas.explore.__MAP_KWARGS`

```python
import geopandas as gpd
import geopandas.explore

geopandas.explore._MAP_KWARGS += ["dragging", "scrollWheelZoom"]

world = gpd.read_file(gpd.datasets.get_path("naturalearth_lowres"))
world.loc[world["continent"].eq("Europe") & ~world["iso_a3"].isin(["RUS","-99"])].explore(
    # m=m,
    column="gdp_md_est",  # make choropleth based on "department" column
    tooltip="name",  # show "decision_type" value in tooltip (on hover)
    popup=True,  # show all values in popup (on click)
    tiles="CartoDB positron",  # use "CartoDB positron" tiles
    cmap="Set1",  # use "Set1" matplotlib colormap
    style_kwds=dict(color="black"),  # use black outline
    zoom_control=False,
    dragging=False,
    scrollWheelZoom=False
)

```

**Comment by martinfleis** [comment]
Thanks! Yes, this is something we need to tackle. The issue is that the keywords you are mentioning are not explicitly defined by folium but passed through kwargs to leaflet. That is why I have missed them when defining `_MAP_KWARGS`. 

I think that we should expose `map_kwds` in `explore` directly to handle these cases. Defining all leaflet args in `_MAP_KWARGS` is not feasible long-term and I don't think we can populate it automatically.

So instead of creating `map_kwds` dictionary here you would update the one (empty by default) coming from args. https://github.com/geopandas/geopandas/blob/cebb5873d04a81bf37376dfb5d4b2adcfe9d87af/geopandas/explore.py#L305

If you'd like to make a PR implementing this, it would be very welcome.

**Comment by rraymondgh** [comment]
I give up,  I've not used a public GitHub repo before.  Have cloned, tested and now trying to commit and push and just getting errors after errors on push.  I can delete ISSUE,  have no way to commit changes and raise a PR

**Comment by jorisvandenbossche** [comment]
Closed by #2315
